### PR TITLE
Change to embedded-graphics-core and update embedded-hal

### PR DESCRIPTION
**Pull request by tarfu**
Use embedded-graphics-core 0.4 and embedded-hal 1.0.0-rc.1

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,13 +10,13 @@ readme = "README.md"
 documentation = "https://docs.rs/st7735-lcd"
 
 [dependencies]
-embedded-hal = "0.2"
-nb = "0.1"
+embedded-hal = "1.0.0-rc.1"
```
Comment (sajattack) [inline Cargo.toml]: I think we should wait until this is stable.  The HALs need time to catch up.

**Review by sajattack — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (clouds56) [reply Cargo.toml]: it is 1.0 now

File: src/lib.rs [hunk 2]
```diff
@@ -117,7 +117,7 @@ where
 
     pub fn hard_reset<DELAY>(&mut self, delay: &mut DELAY) -> Result<(), ()>
     where
-        DELAY: DelayMs<u8>,
+        DELAY: DelayNs,
```
Comment (sajattack) [inline src/lib.rs:120]: What is the reason for changing to DelayNs? I think maybe not all devices support nanosecond resolution and we definitely don't need it here.

Comment (tarfu) [reply src/lib.rs:120]: Because it is the only trait which can be implemented in 1.0.0 if I haven't missed something.

Comment (sajattack) [reply src/lib.rs:120]: I see.

**Review by sajattack — APPROVED** [review]

**Pull request merged**